### PR TITLE
release notes of 1.5.0

### PR DESCRIPTION
**Pull request by kuenishi**
So far this notes includes commits until 37f18eabc4bfc1be1ba01ae58

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
+# Riak CS 1.5.0 Release Notes
+
+## Additions
+
+* a new command `riak-cs-debug` which retrieves diagnosis info includes `cluster-info` [riak_cs/#769](https://github.com/basho/riak_cs/pull/769), [riak_cs/#832](https://github.com/basho/riak_cs/pull/832)
```
Comment (kuenishi) [inline RELEASE-NOTES.md]: s/includes/including/

**Comment by kuenishi** [comment]
As there have been a lot of commits which are not included in release/1.4 but in develop since last August, a few of them might be missing in the notes. Please take a glance and find missing.

**Comment by kuenishi** [comment]
TODO: add leeway seconds notice

**Comment by shino** [comment]
Timestamp in GC buckets  and used by GC daemon is changed. 
It's worth to add upgrading section.
- If I understand correctly, CS node with GC enabled is upgraded first.
- If multiple CS nodes with GC enabled, is it safe to disable GC at all nodes?
- It takes, at most, 2 x leeway seconds to collect gabarges
  (old nodes put with "now + leeway" to the gc bucket, and new GC daemon uses "now - leeway")

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,3 +1,105 @@
+# Riak CS 1.5.0 Release Notes
+
+## Additions
+
+* a new command `riak-cs-debug` which retrieves diagnosis info including `cluster-info` [riak_cs/#769](https://github.com/basho/riak_cs/pull/769), [riak_cs/#832](https://github.com/basho/riak_cs/pull/832)
+* tie up all existing commands into a new command `riak-cs-admin` [riak_cs/#839](https://github.com/basho/riak_cs/pull/839)
+* add a command `riak-cs-stanchion` to switch Stanchion manually [riak_cs/#657](https://github.com/basho/riak_cs/pull/657)
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: switch Stanchion what? Maybe:

add a command `riak-cs-stanchion` to switch Stanchion IP manually

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,3 +1,105 @@
+# Riak CS 1.5.0 Release Notes
+
+## Additions
+
+* a new command `riak-cs-debug` which retrieves diagnosis info including `cluster-info` [riak_cs/#769](https://github.com/basho/riak_cs/pull/769), [riak_cs/#832](https://github.com/basho/riak_cs/pull/832)
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: Minor nitpick: some of these bullets start with a capital letter, and some start with lowercase

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,3 +1,105 @@
...
+* add a command `riak-cs-stanchion` to switch Stanchion manually [riak_cs/#657](https://github.com/basho/riak_cs/pull/657)
+* Performance of garbage collection has been improved via Concurrent GC [riak_cs/#830](https://github.com/basho/riak_cs/pull/830)
+* iterator refresh [riak_cs/#805](https://github.com/basho/riak_cs/pull/805)
+* fold_objects made default in Riak CS [riak_cs/#737](https://github.com/basho/riak_cs/pull/737), [riak_cs/#785](https://github.com/basho/riak_cs/pull/785)
+* Syslog support through lager_syslog [riak_cs/#617](https://github.com/basho/riak_cs/pull/617)
+* Add support for Cache-Control header [riak_cs/#821](https://github.com/basho/riak_cs/pull/821)
+* Allow objects to be reaped sooner than leeway interval. [riak_cs/#470](https://github.com/basho/riak_cs/pull/470)
+* Add option to allow a delay for the initial GC collection [riak_cs/#688](https://github.com/basho/riak_cs/pull/688)
+* Update to lager 2.0.3
+* compiles with R16B0x (Releases still by R15B01)
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: nice

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,3 +1,105 @@
...
+* Setting ACLs via headers at PUT Object creation [riak_cs/#631](https://github.com/basho/riak_cs/pull/631)
+* Improve handling of poolboy timeouts during ping requests [riak_cs/#763](https://github.com/basho/riak_cs/pull/763)
+
+## Notes on Upgrading
+
+### Unaborted or uncompleted multipart uploads
+
+[riak_cs/#475](https://github.com/basho/riak_cs/issues/475) was a
+security issue that flaws remaining multipart uploads, where an newly
+created bucket may include unaborted or uncompleted multipart uploads
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: s/uncompleted/incomplete

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,3 +1,105 @@
...
+which was created in previous epoch of the bucket with same name. This
+was fixed by:
+
+- on creating buckets; checking if live multipart exists and if
+  exists, return 500 failure to client.
```
Comment (reiddraper) [inline RELEASE-NOTES.md:42]: did we end up return 500 here?

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,3 +1,105 @@
...
+* handling empty query string on list multipart upload [riak_cs/#843](https://github.com/basho/riak_cs/pull/843)
+* Setting ACLs via headers at PUT Object creation [riak_cs/#631](https://github.com/basho/riak_cs/pull/631)
+* Improve handling of poolboy timeouts during ping requests [riak_cs/#763](https://github.com/basho/riak_cs/pull/763)
+
+## Notes on Upgrading
+
+### Unaborted or uncompleted multipart uploads
+
+[riak_cs/#475](https://github.com/basho/riak_cs/issues/475) was a
+security issue that flaws remaining multipart uploads, where an newly
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: "security issue that flaws remaining multipart uploads"

I'm not sure what "flaws" is supposed to mean here.

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,3 +1,105 @@
...
+was fixed by:
+
+- on creating buckets; checking if live multipart exists and if
+  exists, return 500 failure to client.
+
+- on deleting buckets; trying to clean up all live multipart remains,
+  and checking if live multipart remains (in stanchion). if exists,
+  return 409 failure to client.
+
+Note that a few operation is needed after upgrading from 1.4.x (or
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: s/is/are

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,3 +1,105 @@
...
+  to cleanup all buckets
+- there might be a time period until above cleanup finished, where no
+  client can create bucket if unfinished multipart upload remains
+  under deleted bucket. You can find [critical] log if such bucket
+  creation is attempted.
+
+### Leeway seconds and disk space
+
+[riak_cs/#470](https://github.com/basho/riak_cs/pull/470) changed the
+behaviour of object deletion and garbage collection. Timestamp in
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: s/Timestamp/The timestamps

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,3 +1,105 @@
...
+garbage collection bucket was changed from the current time when the
```
Comment (reiddraper) [inline RELEASE-NOTES.md]: s/was/were

File: RELEASE-NOTES.md (hunk 6 above)
Comment (kuenishi) [inline RELEASE-NOTES.md:42]: Yes. This is a temporary and abnormal state until cleanup operation mentioned below finishes. And Riak CS should not have multipart uploads in deleted buckets - that's why 500, not 409.

Comment (reiddraper) [inline RELEASE-NOTES.md:42]: sounds good

**Comment by reiddraper** [comment]
Looking good to me. Wonder why the build is failing.

**Comment by kuenishi** [comment]
Thanks, Reid!
We might add a few more bugfix into the list via testing pre-1.5.0.

**Comment by kuenishi** [comment]
@borshop merge

**Pull request merged**